### ISSUE TITLE
unattended_install url is broken

### ISSUE THREAD

**Issue by Andrei-Stepanov** [avocado-framework/avocado-vt#170]
Hello

It seems that we have broken unattended_install url

It worked week ago.

But not now.

```
# avocado run --vt-guest-os Linux.RHEL.7.1.x86_64.i440fx 'unattended_install.url.in_cdrom_ks.default_install.aio_threads' --show-job-log
Config file /mnt/tests/spice/qe-tests/avocado-data/avocado-vt/backends/qemu/cfg/subtests.cfg auto generated from subtest samples
Command line: /mnt/tests/spice/qe-tests/avocado/scripts/avocado run --vt-guest-os Linux.RHEL.7.1.x86_64.i440fx unattended_install.url.in_cdrom_ks.default_install.aio_threads --show-job-log

Avocado version: 0.28.0
Avocado git repo info
Top commit: d12014430451069cbe3e9bac82c82a068c5f1bed
Branch: master

Plugins enabled:
config       Implements the avocado 'config' subcommand
distro       Implements the avocado 'distro' subcommand
exec_path    Implements the avocado 'exec-path' subcommand
gdb          Run tests with GDB goodies enabled
htmlresult   HTML job report
inner_runner Allows the use of an intermediary inner test runner
json         JSON output
journal      Test journal
multiplexer  Implements the avocado 'multiplex' subcommand
plugins_list Implements the avocado 'plugins' subcommand
run_remote   Run tests on a remote machine
sysinfo      Collect system information
test_lister  Implements the avocado 'list' subcommand
test_runner  Implements the avocado 'run' subcommand
vt_bootstrap Avocado VT - implements the 'vt-bootstrap' subcommand
vt.common.arch                          
vt.common.machine_type                  
vt.common.nettype                       
vt.common.netdst                        virbr0
vt.qemu.qemu_bin                        
vt.qemu.qemu_dst_bin                    
vt.qemu.accel                           kvm
vt.qemu.vhost                           off
vt.qemu.monitor                         human
vt.qemu.smp                             2
vt.qemu.image_type                      qcow2
vt.qemu.nic_model                       virtio_net
vt.qemu.disk_bus                        virtio_blk
vt.qemu.sandbox                         on
vt.qemu.defconfig                       yes
vt.qemu.malloc_perturb                  yes
vt.libvirt.connect_uri                  qemu:///session
vt.debug.no_cleanup                     False

Avocado Data Directories:

Avocado replaces config dirs that can't be accessed
with sensible defaults. Please edit your local config
file to customize values

base     /mnt/tests/spice/qe-tests/avocado-base
tests    /mnt/tests/spice/qe-tests/avocado/examples/tests
data     /mnt/tests/spice/qe-tests/avocado-data
logs     /mnt/tests/spice/qe-tests/avocado-logs

Temporary dir: /var/tmp/avocado_XW9OQN

Variant 1:    /

Job ID: 937c6f7399b58a29d9e69933d75fd1256b2fcc6d

START io-github-autotest-qemu.unattended_install.url.in_cdrom_ks.default_install.aio_threads
Avocado-VT '0.28.1-53-ga1982', Branch 'master', SHA1 'a198227c61f5df10e689d86373a15a17f9afe9d3'
Starting test 0
Test parameters:
    _name_map_file = {'tests.cfg': 'qemu_kvm_jeos_quick', 'machines.cfg': 'i440fx', 'subtests.cfg': '(subtest=io-github-autotest-qemu).(subtest=unattended_install).url.in_cdrom_ks.default_install.aio_threads', 'guest-hw.cfg': 'bridge.default_bios.no_virtio_rng.(image_backend=filesystem).no_9p_export.smallpages.no_pci_assignable.qcow2.virtio_blk.smp2.virtio_net', 'guest-os.cfg': 'Linux.RHEL.7.1.x86_64'}
    _short_name_map_file = {'tests.cfg': 'qemu_kvm_jeos_quick', 'machines.cfg': 'i440fx', 'subtests.cfg': 'io-github-autotest-qemu.unattended_install.url.in_cdrom_ks.default_install.aio_threads', 'guest-hw.cfg': 'bridge.default_bios.no_virtio_rng.filesystem.no_9p_export.smallpages.no_pci_assignable.qcow2.virtio_blk.smp2.virtio_net', 'guest-os.cfg': 'Linux.RHEL.7.1.x86_64'}
    auto_cpu_model = yes
    avocado_inject_params = True
    avocado_params = ([TreeNode(name='')], ['/run/*'])
    backup_dir = images/
    backup_image = yes
    backup_image_before_testing = yes
    backup_image_on_check_error = no
    boot_menu = off
    boot_once = c
    boot_order = cdn
    boot_path = images/pxeboot
    catch_monitor = catch_monitor
    cd_format = ide
    cdrom_check_cdrom_pattern = /dev/cdrom-\w+|/dev/cdrom\d*
    cdrom_get_cdrom_cmd = ls /dev/cdrom*
    cdrom_info_cmd = cat /proc/sys/dev/cdrom/info
    cdrom_test_cmd = dd if=%s of=/dev/null bs=1 count=1
    cdrom_unattended = images/rhel71-64/ks.iso
    cdroms = cd1
    check_image = yes
    cmds_installed_host = 
    connect_uri = default
    convert_ppm_files_to_png = no
    cpu_chk_cmd = grep -c processor /proc/cpuinfo
    cpu_model_flags = 
    dep = []
    display = vnc
    drive_format = virtio
    drive_index_cd1 = 1
    drive_index_image1 = 0
    drive_rerror_image1 = 
    drive_werror_image1 = 
    enable_msix_vectors = yes
    enable_pvpanic = yes
    encode_video_files = yes
    file_transfer_client = scp
    file_transfer_port = 22
    force_create_image = yes
    grub_file = /boot/grub2/grub.cfg
    guest_port_remote_shell = 22
    guest_port_unattended_install = 12323
    id = io-github-autotest-qemu.unattended_install.url.in_cdrom_ks.default_install.aio_threads
    image_aio = threads
    image_backend = filesystem
    image_boot = yes
    image_chain = 
    image_clone_command = cp --reflink=auto %s %s
    image_copy_on_error = no
    image_format = qcow2
    image_name = images/rhel71-64
    image_raw_device = no
    image_remove_command = rm -rf %s
    image_size = 10G
    image_unbootable_pattern = Hard Disk.*not a bootable disk
    image_verify_bootable = no
    images = image1
    inactivity_treshold = 1800
    inactivity_watcher = error
    indirect_image_blacklist = /dev/hda[\d]* /dev/sda[\d]* /dev/sg0 /dev/md0
    initrd = images/rhel71-64/initrd.img
    install_virtio = no
    ioport_pvpanic = 0x505
    keep_ppm_files = no
    keep_ppm_files_on_error = no
    keep_screendumps = yes
    keep_screendumps_on_error = yes
    keep_video_files = yes
    keep_video_files_on_error = yes
    kernel = images/rhel71-64/vmlinuz
    kernel_params = ks=cdrom nicdelay=60  console=ttyS0,115200 console=tty0
    kill_timeout = 60
    kill_unresponsive_vms = yes
    kill_vm = yes
    kill_vm_before_test = no
    kill_vm_gracefully = yes
    kill_vm_on_error = yes
    local_ip = ENTER.YOUR.LOCAL.EXAMPLE.COM
    local_pwd = 
    login_timeout = 360
    lowest_mem = 512
    machine_type = pc
    main_vm = avocado-vt-vm1
    malloc_perturb = yes
    medium = url
    unattended_file = unattended/RHEL-7-series.ks
    url = http:// CUTTED OFF
    url_auto_ip = 192.168.122.1
    usb_bus = usb1.0
    usb_controller = ehci
    usb_devices = tablet1
    usb_max_port = 6
    usb_type = ich9-usb-ehci1
    usb_type_tablet1 = usb-tablet
    usb_type_usb1 = ich9-usb-ehci1
    usbs = usb1
    use_autostart = no
    use_check_cpu = yes
    use_debug = yes
    use_no_reboot = no
    use_os_type = yes
    use_os_variant = no
    use_virt_install_wait = no
    used_cpus = 2
    used_mem = 512
    username = root
    vcpu_thread_pattern = thread_id.?[:|=]\s*(\d+)
    verify_host_dmesg = yes
    vfd_size = 1440k
    vga = std
    virt_install_wait_time = 300
    virt_test_type = qemu libvirt
    vm_arch_name = x86_64
    vm_register_delay = 5
    vm_type = qemu
    vms = avocado-vt-vm1
    vnc_autoport = yes

Testing with nettype='user' while running as root may produce unexpected results!!!

Creating new, empty env file
Searching for test modules that match 'type = unattended_install' and 'provider = io-github-autotest-qemu' on this cartesian dict
Found subtest module /mnt/tests/spice/qe-tests/avocado-data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/generic/tests/unattended_install.py
Detected CPU vendor as 'AuthenticAMD'
KVM version: 3.10.0-229.el7.x86_64
KVM userspace version: 1.5.3 (qemu-kvm-1.5.3-86.el7)
Running 'true'
Image backup /mnt/tests/spice/qe-tests/avocado-data/avocado-vt/images/rhel71-64.qcow2.backup already exists, skipping...
Create image on local fs.
Context: preprocessing --> Create image by command: /bin/qemu-img create -f qcow2 /mnt/tests/spice/qe-tests/avocado-data/avocado-vt/images/rhel71-64.qcow2 10G
avocado-vt-vm1 is not alive. Can't query the register status
Starting unattended install setup
Running 'cp -r /var/tmp/avocado_XW9OQN/cdrom_cd1_3RJHZE/isolinux/ /var/tmp/avocado_XW9OQN/cdrom_unattended_MSVPsd/'
[stderr] cp: cannot stat ‘/var/tmp/avocado_XW9OQN/cdrom_cd1_3RJHZE/isolinux/’: No such file or directory
Checking image file /mnt/tests/spice/qe-tests/avocado-data/avocado-vt/images/rhel71-64.qcow2
Image backup /mnt/tests/spice/qe-tests/avocado-data/avocado-vt/images/rhel71-64.qcow2.backup already exists, skipping...
Copying /mnt/tests/spice/qe-tests/avocado-data/avocado-vt/images/rhel71-64.qcow2 -> /mnt/tests/spice/qe-tests/avocado-data/avocado-vt/images/rhel71-64.qcow2.backup
Thread quit. Used to failed to get register info from guest 20151001-111739-BSBdm94T for 2 times.
Running 'true'

Reproduced traceback from: /mnt/tests/spice/qe-tests/avocado/avocado/core/plugins/vt.py:575
Traceback (most recent call last):
  File "/mnt/tests/spice/qe-tests/avocado/avocado/core/plugins/vt.py", line 361, in runTest
    self._runTest()
  File "/mnt/tests/spice/qe-tests/avocado/avocado/core/plugins/vt.py", line 494, in _runTest
    run_func(self, params, env)
  File "/mnt/tests/spice/qe-tests/avocado-data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/generic/tests/unattended_install.py", line 14, in run
    unattended_install.run(test, params, env)
  File "/mnt/tests/spice/qe-tests/avocado/virttest/error_context.py", line 135, in new_fn
    return fn(*args, **kwargs)
  File "/mnt/tests/spice/qe-tests/avocado/virttest/tests/unattended_install.py", line 1105, in run
    unattended_install_config.setup()
  File "/mnt/tests/spice/qe-tests/avocado/virttest/tests/unattended_install.py", line 985, in setup
    self.setup_boot_disk()
  File "/mnt/tests/spice/qe-tests/avocado/virttest/tests/unattended_install.py", line 672, in setup_boot_disk
    kernel_params)
  File "/mnt/tests/spice/qe-tests/avocado/virttest/utils_disk.py", line 368, in __init__
    process.run(cp_cmd)
  File "/mnt/tests/spice/qe-tests/avocado/avocado/utils/process.py", line 949, in run
    raise CmdError(cmd, sp.result)
CmdError: Command 'cp -r /var/tmp/avocado_XW9OQN/cdrom_cd1_3RJHZE/isolinux/ /var/tmp/avocado_XW9OQN/cdrom_unattended_MSVPsd/' failed (rc=1)

Traceback (most recent call last):

  File "/mnt/tests/spice/qe-tests/avocado/avocado/core/plugins/vt.py", line 617, in _run_avocado
    raise test_exception

CmdError: Command 'cp -r /var/tmp/avocado_XW9OQN/cdrom_cd1_3RJHZE/isolinux/ /var/tmp/avocado_XW9OQN/cdrom_unattended_MSVPsd/' failed (rc=1)

ERROR io-github-autotest-qemu.unattended_install.url.in_cdrom_ks.default_install.aio_threads -> CmdError: Command 'cp -r /var/tmp/avocado_XW9OQN/cdrom_cd1_3RJHZE/isolinux/ /var/tmp/avocado_XW9OQN/cdrom_unattended_MSVPsd/' failed (rc=1)

Test results available in /mnt/tests/spice/qe-tests/avocado-logs/job-2015-10-01T11.17-937c6f7

```

**Comment by lmr** [comment]
It'd be helpful to get some sort of bisect. When you said it was working, getting a last known working commit would be extremely useful.

That said, I don't think you are supposed to use the `in_cdrom` variant. If I recall correctly, that variant is only for when you have an install media that ships with an unattended file (in this case, a kickstart).

**Comment by Andrei-Stepanov** [comment]
I agree.
I tried to run it on earlier version of avocado/avocado-vt.
But it fails with the same error.
I am quite sure I was able use **url**
I wasn't  be able to find commit that broken our installation.
If you wish I can investigate more.

**Comment by lmr** [comment]
I'm investigating, trying with the Fedora 21 install. There seems to be something broken, but I can't tell what it is. Anaconda says it can't access the kickstart file, yet, the file is there when you look in the command line from inside the terminal in the initrd image. If you could investigate, that'd be nice.

**Comment by clebergnu** [comment]
Without taking a detailed look, based on your description, it looks like some kind of network issues between the guest and the (host side) HTTP server that supplies the Kickstart file. Just a guess though.

**Comment by lmr** [comment]
@clebergnu Yep, that's what I thought. But he's using a variant that doesn't get the ks file from the http server. I tried with the http_ks variant, and what I found (even after `iptables -F`) was that the VM could not get the kickstart.

**Comment by lmr** [comment]
![screenshot from 2015-10-01 09-47-51](https://cloud.githubusercontent.com/assets/296807/10220979/9395fd72-6821-11e5-825b-212bc5653d77.png)

@clebergnu So the screenshot above is what I'm getting. I'm looking at the code to see where it gets that IP from. It doesn't seem to be reachable.

**Comment by lmr** [comment]
The funny part is that if you select the variant that gets the kickstart from a CD that we create, Anaconda can't get to the kickstart file either, even though I was able to mount the cdrom file, verify it has the correct label and the kickstart file. This is puzzling.

**Comment by lmr** [comment]
Although I guess we should be using bridge instead of user mode if we want to use the http server.

**Comment by lmr** [comment]
![screenshot from 2015-10-01 10-32-35](https://cloud.githubusercontent.com/assets/296807/10221965/dd986eb8-6827-11e5-88bb-49dc9f2d14d5.png)

After fixing up some things, I get to this point. For some reason, `ks.cfg` (the correct location) is being merged with `-label/CDROM` (I'm not sure where this is getting appened to that string). I'll keep looking.

**Comment by lmr** [comment]
@Andrei-Stepanov, after investigation, I'm pretty sure uou are selecting the wrong unattended file delivery method. `in_cdrom_ks` is intended for install medias that have the unattended file right into them, not the case for RHEL7 and many other guest OS.

For the http and unattended cdrom methods, I found one fix and developed a few other conveniences for people running those tests frequently. If you could please try them out, that'd be great. Thank you.

**Comment by lmr** [comment]
https://github.com/avocado-framework/avocado-vt/pull/171

**Comment by lmr** [comment]
Please try out the PR above and let me know of the results.

**Comment by Andrei-Stepanov** [comment]
Hello
It fails with:
```
File "/mnt/tests/spice/qe-tests/avocado/virttest/tests/unattended_install.py", line 902, in setup_url
    sha1sum_kernel = sha1sum_kernel_output.split()[0]
IndexError: list index out of range
```

**Comment by lmr** [comment]
I'm closing this one as the problem has been fixed with #171 and subsequent patches.
